substrate/frame/utility/src/lib.rs

```diff
@@ -120,6 +120,10 @@ pub mod pallet {
 		ItemFailed { error: DispatchError },
 		/// A call was dispatched.
 		DispatchedAs { result: DispatchResult },
+		/// one or both if_else calls failed.
+		IfElseFailure { which: String, error: DispatchError },
```

Comment:
no string in pallets, the current implementation make it hard for sdk users to figure out the meaning of this
it could be an enum

but I don’t think the current event make sense as there are only 3 possibilities: main success, mail failed + fallback success, all failed
the all failed case should just result the whole call to be failed. ie no extra event. so we only really two event: main success, and fallback called